pennylane/gradients/hadamard_gradient.py

```diff
@@ -304,8 +304,10 @@ def _expval_hadamard_grad(tape, argnum, aux_wire):
 
             new_tape = qml.tape.QuantumScript(ops=ops, measurements=measurements)
 
-            new_tape.expand()
+            _rotations, _measurements = qml.tape.tape.rotations_and_diagonal_measurements(new_tape)
```

Comment:
the old behaviour here is funny, and the reason for the changes to `tape.py`. Calling `new_tape.expand()` _returns_ a newly created tape with the rotations and diagonal measurements, but it ALSO used to update the inputted tape (`new_tape`) to have them as well, and that was not good. I could say `new_tape = new_tape.expand()`, but it might do other expansion that we don't want. The original purpose of this line here was to (sneakily) update `new_tape` with rotations and diagonal measurements, so I'm now doing that here (at the caller) along with the helper function I've added.

Reply:
Thanks!